File: OCH_Tao/ABC/B/277.js
const S = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(S.shift(), 10);
const RE = new RegExp("^[HDCS][A23456789TJQK]$");
console.log(new Set(S.filter(x => RE.test(x))).size === N ? "Yes" : "No");

File: OCH_Tao/ABC/B/277.py
import re
N = int(input())
L = set()
R = re.compile(r"[HDCS][A23456789TJQK]")
for i in range(N):
  S = input()
  if re.fullmatch(R,S):
    L.add(S)
print("Yes" if len(L)==N else "No")